From 66f1aaa8227bd7784c3e451b6463fc20fdabefe0 Mon Sep 17 00:00:00 2001
From: Saurabh Sengar <saurabh.singh@xilinx.com>
Date: Sat, 5 May 2018 06:14:56 +0530
Subject: [PATCH 0352/1852] drm: xlnx: mixer: making logo layer optional

commit 0af50bc53c9ab09b2130db6e15d6255c45b9b3be from
https://github.com/Xilinx/linux-xlnx.git

There are two things in mixer device tree.
1. xlnx,logo-layer property
2. logo node

So this leads to two scenarios:
1. If xlnx,logo-layer is there then driver is expected to parse logo node
2. If xlnx,logo-layer is not there then driver shouldn't look for logo node
Issue was driver code was looking for logo node when xlnx,logo-layer
was not there, and hence the bug, which is fixed by this patch.

Signed-off-by: Saurabh Sengar <saurabh.singh@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_mixer.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_mixer.c b/drivers/gpu/drm/xlnx/xlnx_mixer.c
index f2a08b394beb..49c896c5ed75 100644
--- a/drivers/gpu/drm/xlnx/xlnx_mixer.c
+++ b/drivers/gpu/drm/xlnx/xlnx_mixer.c
@@ -2062,9 +2062,11 @@ static int xlnx_mix_dt_parse(struct device *dev, struct xlnx_mix *mixer)
 	ret = xlnx_mix_parse_dt_bg_video_fmt(node, mixer_hw);
 	if (ret)
 		return ret;
-	/* read logo data from dts */
-	ret = xlnx_mix_parse_dt_logo_data(node, mixer_hw);
+	if (mixer_hw->logo_layer_en) {
+		/* read logo data from dts */
+		ret = xlnx_mix_parse_dt_logo_data(node, mixer_hw);
 		return ret;
+	}
 	return 0;
 }
 
-- 
2.31.1

